trace-utils/src/stats_utils.rs

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use flate2::{write::GzEncoder, Compression};
+use hyper::{body::Buf, Body, Client, Method, Request, StatusCode};
+use hyper_rustls::HttpsConnectorBuilder;
+use log::debug;
+use std::{env, io::Write};
+
+use datadog_trace_protobuf::pb;
+
+const STATS_INTAKE_URL: &str = "https://trace.agent.datadoghq.com/api/v0.2/stats";
+
+pub async fn get_stats_from_request_body(body: Body) -> anyhow::Result<pb::ClientStatsPayload> {
+    let buffer = hyper::body::aggregate(body).await?;
+
+    let client_stats_payload: pb::ClientStatsPayload = match rmp_serde::from_read(buffer.reader()) {
+        Ok(res) => res,
+        Err(err) => {
+            anyhow::bail!("Error deserializing stats from request body: {err}")
+        }
+    };
+
+    if client_stats_payload.stats.is_empty() {
+        debug!("Empty trace stats payload received");
+        anyhow::bail!("No stats in stats payload");
+    }
+    Ok(client_stats_payload)
+}
+
+pub fn construct_stats_payload(stats: Vec<pb::ClientStatsPayload>) -> pb::StatsPayload {
+    pb::StatsPayload {
+        agent_hostname: "".to_string(),
+        agent_env: "".to_string(),
+        stats,
+        agent_version: "".to_string(),
+        client_computed: true,
+    }
+}
+
+pub fn serialize_stats_payload(payload: pb::StatsPayload) -> anyhow::Result<Vec<u8>> {
+    let msgpack = rmp_serde::to_vec_named(&payload)?;
+    let mut encoder = GzEncoder::new(Vec::new(), Compression::fast());
+    encoder.write_all(&msgpack)?;
+    match encoder.finish() {
+        Ok(res) => Ok(res),
+        Err(e) => anyhow::bail!("Error serializing stats payload: {e}"),
+    }
+}
+
+pub async fn send_stats_payload(data: Vec<u8>) -> anyhow::Result<()> {
+    let api_key = match env::var("DD_API_KEY") {
```

Comment:
In telemetry crate - we already have some code to parse env variables - including the ability to set DD_SITE 
 - etc https://github.com/DataDog/libdatadog/blob/main/ddtelemetry/src/config.rs
 
Might be worth extracting this code into some common crate and attempting a reuse.

Reply:
Additionally I would recommend not using global variable like `env::var` inside the code here. Its generally better to create a `Config` object - that is constructed from env. To make the code more testable

Reply:
moved api_key to a config struct